File: gui/src/dtos/requests/ClientRequestDTO.ts
/**
 * Client request DTO.
 *
 * @property {number} id - The ID of the client
 * @property {string} firstName - The first name of the client
 * @property {string} lastName - The last name of the client
 * @property {string} email - The email of the client
 * @property {string} phone - The phone number of the client
 * @property {string} uniqueIdentifier - The unique identifier of the client
 */
export interface ClientRequestDTO {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  uniqueIdentifier: string;
}

File: api/src/main/java/com/sinergy/chronosync/dto/request/ClientRequestDTO.java
package com.sinergy.chronosync.dto.request;

import com.sinergy.chronosync.model.Client;
import com.sinergy.chronosync.model.Firm;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for creating or updating a client.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ClientRequestDTO {

	private Long id;
	private String firstName;
	private String lastName;
	private String email;
	private String phone;
	private String address;
	private Long firmId;

	/**
	 * Creates and returns Client model from Data Transfer Object.
	 *
	 * @return {@link Client} user model
	 */
	public Client toModel() {
		Firm firm = new Firm();
		firm.setId(firmId);

		return Client.builder()
				.id(id)
				.firstName(firstName)
				.lastName(lastName)
				.email(email)
				.phone(phone)
				.address(address)

				.firm(firm)
				.build();
	}

}
